jobs/pacman-data-shipper/src/main/java/com/tmobile/cso/pacman/datashipper/util/HttpUtil.java

```diff
@@ -124,32 +122,44 @@ public boolean isTrusted(X509Certificate[] arg0, String arg1) throws Certificate
                         }
                     }).build()).build();
         } catch (KeyManagementException | NoSuchAlgorithmException | KeyStoreException e) {
-            LOGGER.error("Error getting getHttpClient " , e);
+            LOGGER.error("Error getting getHttpClient ", e);
         }
+
         return httpClient;
     }
-    
-    public static String httpGetMethodWithHeaders(String url,Map<String, Object> headers) throws Exception {
+
+    public static String httpGetMethodWithHeaders(String url, Map<String, Object> headers) throws Exception {
         String json = null;
-        
         HttpGet get = new HttpGet(url);
-        CloseableHttpClient httpClient = null;
+
         if (headers != null && !headers.isEmpty()) {
             for (Map.Entry<String, Object> entry : headers.entrySet()) {
                 get.setHeader(entry.getKey(), entry.getValue().toString());
             }
         }
-        try {
-            httpClient = getHttpClient();
+
+        try (CloseableHttpClient httpClient = getHttpClient()) {
             CloseableHttpResponse res = httpClient.execute(get);
             if (res.getStatusLine().getStatusCode() == 200) {
                 json = EntityUtils.toString(res.getEntity());
             }
-        } finally {
-            if (httpClient != null) {
-                httpClient.close();
-            }
         }
+
         return json;
     }
+
+    // Method to generate the base URL for the asset service
+    public static String getAssetServiceBaseUrl() {

```

Comment:
these should be part of the service and not http utiils. http utils should be simply getting data